lib/Sema/CSDiag.cpp

```diff
@@ -6048,13 +6053,14 @@ bool FailureDiagnosis::visitAssignExpr(AssignExpr *assignExpr) {
 
   auto destType = destExpr->getType();
   if (destType->is<UnresolvedType>() || destType->hasTypeVariable()) {
-    // If we have no useful type information from the destination, just type
-    // check the source without contextual information.  If it succeeds, then we
-    // win, but if it fails, we'll have to diagnose this another way.
-    return !typeCheckChildIndependently(assignExpr->getSrc());
+    // Look closer into why destination has unresolved types since such
+    // means that destination has diagnosible structural problems, and it's
+    // better to diagnose destination (if possible) before moving on to
+    // the source of the assignment.
+    return !typeCheckChildIndependently(
+        destExpr, TCC_AllowLValue | TCC_ForceRecheck, false);
```

Comment:
In this case it doesn't make much sense to look at source because destination is going to produce diagnostics since there were unresolved type variables involved in previous solution.